cachelib/allocator/CacheAllocator.h

```diff
@@ -2051,6 +2051,8 @@ auto& mmContainer = getMMContainer(tid, pid, cid);
         // TODO: only allow it for read-only items?
         // or implement mvcc
         if (candidate->markMoving(true)) {
+          // promotions should rarely fail since we already marked moving
+          mmContainer.remove(itr);
```

Comment:
that won't work for chained items, right? Maybe put a 'todo' here?

Reply:
We already check if it is a chained item and will throw exception if it is.